sourcegraph/sourcegraph
# feat(setup-checklist): add site config license form

Closes https://github.com/sourcegraph/sourcegraph/issues/56002.

This PR:
* add site config license form
* add `licenseInfo` GQL API for parsing license info from license key
* add `useJSONCParser` hook to parse/update/reset JSON using `jsonc-parser`
* refactor `SMTPConfigForm` to use new `useJSONCParser` hook
* move `SMTPConfigForm` to "src/site-admin/site-config/forms/smtp" similar to ".../forms/license"
* refactor `SiteAdminConfigurationPage` errors rendering

## Test plan
- `EXTSVC_CONFIG_ALLOW_EDITS=true sg start`
- Enable `setup-checklist` feature flag if not yet enabled
- Open https://sourcegraph.test:3443/site-admin/configuration/basic
- Check that license form renders current license info correctly
- Update license key and check that license info is recalculated/updated
- Save config via "Update site configuration" button. Reload page and check that license has successfully saved

**Known issues:
- Form changes are not reflected in JSON editor. It requires another set of changes to `DynamicallyImportedMonacoSettingsEditor` to make reflect on the changed props. This will be a separate PR

## Screenshots 
<img width="929" alt="image" src="https://github.com/sourcegraph/sourcegraph/assets/6717049/54855b3a-1bc1-4702-82d0-2e666b88bb3f">
<img width="929" alt="image" src="https://github.com/sourcegraph/sourcegraph/assets/6717049/dd392a12-a03e-47da-8353-e8d6130685d2">
